web/src/main/java/com/epam/rft/atsy/web/controllers/CandidateCreationController.java

```diff
@@ -30,11 +31,16 @@
    * @param candidateId the identifier of the candidate whose data should be given back
    * @return a ModelAndView object which contains the all candidate information and the name of the
    * page that handles the candidate data
+   * @throws com.epam.rft.atsy.web.exceptionhandling.CandidateNotFoundException
    */
   @RequestMapping(method = RequestMethod.GET, path = "/{candidateId}")
-  public ModelAndView loadCandidate(@PathVariable(value = "candidateId") Long candidateId) {
+  public ModelAndView loadCandidate(@PathVariable(value = "candidateId") Long candidateId) throws CandidateNotFoundException {
     ModelAndView modelAndView = new ModelAndView(VIEW_NAME);
-    modelAndView.addObject(CANDIDATE_OBJECT_KEY, candidateService.getCandidate(candidateId));
+    CandidateDTO candidate = candidateService.getCandidate(candidateId);
+    if (candidate == null)
+      throw new CandidateNotFoundException();
```

Comment:
Please add technical details to the exception, here the referenced `candidateId` might be useful to be placed into the exception.
If the future you will need to find out why so much 404 errors are sent back to the users, then you can see not just the candidate was not found, but which candidates id were referenced.